File: backend/app/schemas/project.py
from pydantic import BaseModel, UUID4
from typing import Optional, List, Dict, Any
from datetime import datetime
from uuid import UUID
from .task import TaskResponse

class ProjectBase(BaseModel):
    title: str
    description: Optional[str] = None
    status: Optional[str] = "active"

class ProjectCreate(ProjectBase):
    pass

class ProjectUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    status: Optional[str] = None

class ProjectResponse(ProjectBase):
    id: UUID
    user_id: Optional[UUID] = None
    created_at: datetime
    updated_at: datetime
    tasks: Optional[List[TaskResponse]] = []

    class Config:
        from_attributes = True
        # Adicione para não carregar relacionamentos automaticamente
        orm_mode = True

File: frontend/app/types/project.ts
import { Task } from './task';

export type ProjectStatus = 'active' | 'completed' | 'archived';

export interface Project {
  id?: string;
  title: string;
  description?: string | null;
  status?: ProjectStatus;
  user_id?: string;
  tasks?: string[]; // Array de IDs de tasks associadas
  created_at?: string;
  updated_at?: string;
}

export type CreateProjectInput = Omit<Project, 'id' | 'user_id' | 'created_at' | 'updated_at'>;

export interface ProjectFilters {
  status?: ProjectStatus;
  searchTerm?: string;
}